src/sparse/node.rs

```diff
@@ -40,17 +48,46 @@ impl Node {
         node
     }
 
-    pub fn create_node(left_child: &Node, right_child: &Node) -> Self {
+    pub fn create_node(left_child: &Node, right_child: &Node, height: u32) -> Self {
```

Comment:
Hmmm. Do we really want to make height a parameter? What's the benefit of not computing it automatically?

Reply:
I use a height parameter here because, as it turns out, creating a node requires context: what are the nodes that we are joining? Height is determined by this context. When joining two leaves, the parent height is calculated from the common path length; when joining nodes, the parent height is the greater height of the two nodes plus 1.  Rather than gathering this context here, I simply allow the caller to calculate these things and pass that info here; this function then becomes a lower level utility while [create_node_on_path](https://github.com/FuelLabs/fuel-merkle/pull/76/files#diff-9ed365b0915815caf8081101aad3ca005ebf811e255ea4ec0b48cefe8846790bR61) becomes the higher level function that determines the required context.

Reply:
And yes - this could certainly be a `u16`. The max height is actually exactly `256`. But keep in mind, that actually represents 257 height values! That's because we have a leaf row with height 0, and 256 internal rows. 

I could change this to use 2 bytes, but the `Node` buffer itself currently uses 4 bytes to store height. I would like to refactor both these things simultaneously in a separate PR.